spec/Section 6 -- Execution.md

```diff
@@ -124,15 +128,34 @@ An initial value may be provided when executing a query.
 
 ExecuteQuery(query, schema, variableValues, initialValue):
 
+  * Let {subsequentPayloads} be an empty list.
   * Let {queryType} be the root Query type in {schema}.
   * Assert: {queryType} is an Object type.
   * Let {selectionSet} be the top level Selection Set in {query}.
   * Let {data} be the result of running
-    {ExecuteSelectionSet(selectionSet, queryType, initialValue, variableValues)}
+    {ExecuteSelectionSet(selectionSet, queryType, initialValue, variableValues, subsequentPayloads)}
     *normally* (allowing parallelization).
   * Let {errors} be any *field errors* produced while executing the
     selection set.
-  * Return an unordered map containing {data} and {errors}.
+  * Yield an unordered map containing {data}, {errors}, and {hasNext} if 
+    {subsequentPayloads} is non-empty.
+  * While {subsequentPayloads} is not empty:
+    * Let {record} be the first complete item in {subsequentPayloads}.
+    * Remove {record} from {subsequentPayloads}.
+    * If {record} is a Deferred Fragment Record:
+      * Let {payload} be the result of running {ResolveDeferredFragmentRecord(record, variableValues, subsequentPayloads)}.
+    * If {record} is a Stream Record:
+      * Let {payload} be the result of running ResolveStreamRecord(record, variableValues, subsequentPayloads).
+    * If {payload} is {null}:
+      * If {subsequentPayloads} is empty:
+        * Yield a map containing a field `hasNext` with the value {false};
+      * If {subsequentPayloads} is not empty:
+        * Continue to the next record in {subsequentPayloads}.
+    * If {record} is not the final element in {subsequentPayloads}
+      * Add an entry to {payload} named `hasNext` with the value {true}.
+    * If {record} is the final element in {subsequentPayloads}
+      * Add an entry to {payload} named `hasNext` with the value {false}.
+    * Yield {payload}
```

Comment:
This case where we yield a payload of only `{ hasNext: false }` can only occur when an async iterable (or equivalent) stream record is the last value in subsequent payloads. I tried to cover it by adding this to `ResolveStreamRecord`:

  * Wait for the next item from {iterator}.
  * If an item is not retrieved because {iterator} has completed: // (I just updated this line to clarify it)
    * Return {null}
  * Let {item} be the item retrieved from {iterator}.
  * // rest of logic
  * Append {streamRecord} to {subsequentPayloads}.
  * Return {payload}.

So unless we know the underlying iterable from the stream has completed, it will be re-added to `{subsequentPayloads}` before it's processed again.

Then, the `{null}` case is handled above this section:

* If {payload} is {null}:
      * If {subsequentPayloads} is empty:
        * Yield a map containing a field `hasNext` with the value {false};
      * If {subsequentPayloads} is not empty:
        * Continue to the next record in {subsequentPayloads}.

Reply:
I did miss adding this to the mutation and subscription section, which I will update

Reply:
I extracted this out to a function `YieldSubsequentPayloads` which is referenced in `ExecuteQuery`, `ExecuteMutation` and `ExecuteSubscriptionEvent`